index.html

```diff
@@ -2084,6 +2116,40 @@ <h2>
           </li>
           <li>Set |response|.{{PaymentResponse/[[complete]]}} to true.
           </li>
+          <li class="addition proposed">Let |serializedData| be the result of
+          <a>JSON-serializing</a> |details|.{{PaymentCompleteDetails/data}}
+          into a string.
+          </li>
+          <li class="addition proposed">If serializing [=exception/throws=] an
+          exception, return <a>a promise rejected with</a> that exception.
+            <div class="issue">
+              What should happen if an exception is thrown here? Should we
```

Comment:
I think we might want to wait to set `response.[[complete]]` to `true` until after we JSON serialize, re-parse, and convert to IDL so that if something happens we can re-call `complete()` (as otherwise the step 2 would always fail).

Also, I wonder if we should wait to set `response.[[complete]]` until right before we resolve the returned promise so that if for some reason the user agent finds issue with the data when it attempts to close the payment UI then the developer can still recover (just like the above)?

Reply:
Agree on point 1 (being able to recover). My worry was that an unhandled promise rejection would leave the payment sheet hanging open, but the browser can either auto-close it after some time or provide some other means for the user to close it. 

About point 2, we could add some thing state "processing" or whatever, but it might be best to hold off until there is a concrete situation where that might happen. In theory, doing the IDL conversion steps should catch any issues... oh! That reminds me, we should do `.data` validation at this point also (same as we did in https://github.com/w3c/payment-request/pull/976).